integration/tests/conftest.py

```diff
@@ -742,6 +742,105 @@ def precompiled_contract(web3_client, faucet, accounts):
     return contract
 
 
+@pytest.fixture(scope="class")
+def chain_execution_contracts(accounts, web3_client):
+    sender_account = accounts[0]
+
+    # Deploy contracts without dependencies first
+    func2, _ = web3_client.deploy_and_get_contract(
```

Comment:
it will be more readable to rename funcs to contracts everywhere when you mean contracts